pkg/eden/vbox.go

```diff
@@ -123,89 +77,173 @@ func StartEVEVBox(vmName, eveImageFile string, cpus int, mem int, hostFwd map[st
-		for k, v := range hostFwd {
-			hostPort, err := strconv.Atoi(k)
-			if err != nil {
-				log.Errorf("Parsing %s to Integer value failed", k)
-				break
-			}
-			hostPort += 10
-			guestPort, err := strconv.Atoi(v)
+// getGuestIPsVBox returns IP addresses allocated to the EVE VM by the VBox DHCP server.
+func getGuestIPsVBox(vmName string) (ipAddrs [2]string, err error) {
+	var output string
+	// First get MAC addresses assigned to EVE's uplink interfaces.
+	output, _, err = utils.RunCommandAndWait("VBoxManage",
+		strings.Fields(fmt.Sprintf("showvminfo %s", vmName))...)
+	if err != nil {
+		return
+	}
+	scanner := bufio.NewScanner(bytes.NewReader([]byte(output)))
+	nicInfoReg := regexp.MustCompile(`NIC ([0-9]+):.* MAC: ([0-9A-F]+),`)
+	var macAddrs [2]string
+	for scanner.Scan() {
+		match := nicInfoReg.FindStringSubmatch(scanner.Text())
+		if len(match) == 3 {
+			nicIdx, err := strconv.Atoi(match[1])
 			if err != nil {
-				log.Errorf("Parsing %s to Integer value failed", v)
-				break
+				continue
 			}
-			guestPort += 10
-			commandArgsString := fmt.Sprintf("natnetwork  modify --netname %s --port-forward-4 :tcp:[]:%d:[%s]:%d",
-											"natnet1", hostPort, eth1IP.String(), guestPort)
-
-			if err = utils.RunCommandWithLogAndWait("VBoxManage", defaults.DefaultLogLevelToPrint, strings.Fields(commandArgsString)...); err != nil {
-				if err.Error() == "exit status 2" {
-					log.Infof("VBoxManage NAT rule: %s: already exists", commandArgsString)
-				} else {
-					log.Fatalf("VBoxManage error for command %s %s", commandArgsString, err)
-				}
+			if nicIdx == 1 || nicIdx == 2 {
+				macAddrs[nicIdx-1] = match[2]
 			}
 		}
-		if poweroff {
-			commandArgsString := fmt.Sprintf("startvm  %s", vmName)
-			if err = utils.RunCommandWithLogAndWait("VBoxManage", defaults.DefaultLogLevelToPrint, strings.Fields(commandArgsString)...); err != nil {
-				log.Fatalf("VBoxManage error for command %s %s", commandArgsString, err)
+	}
+	// Get IP address(es) from DHCP leases associated with EVE's MAC addresses.
+	for i, macAddr := range macAddrs {
+		if macAddr == "" {
+			err = fmt.Errorf("failed to get MAC address of eth%d", i)
+			return
+		}
+		output, _, err = utils.RunCommandAndWait("VBoxManage",
+			strings.Fields(fmt.Sprintf("dhcpserver findlease --network %s --mac-address %s",
+				natNetworkName, macAddr))...)
+		if err != nil {
+			return
+		}
+		scanner := bufio.NewScanner(bytes.NewReader([]byte(output)))
+		const ipAddrPrefix = "IP Address:"
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.HasPrefix(line, ipAddrPrefix) {
+				ipAddrs[i] = strings.TrimPrefix(line, ipAddrPrefix)
+				ipAddrs[i] = strings.TrimSpace(ipAddrs[i])
+				break
 			}
 		}
 	}
-
-	return err
+	for i, ipAddr := range ipAddrs {
+		if ipAddr == "" {
+			err = fmt.Errorf("failed to get IP address of eth%d", i)
+			return
+		}
+	}
+	return
 }
 
 //StopEVEVBox function stop EVE in VirtualBox
 func StopEVEVBox(vmName string) (err error) {
```

Comment:
We will need to make  `VBoxManage dhcpserver remove ` 
Otherwise, we will get following error when we will start again 
```
VBoxManage: error: Cannot start DHCP server because it is already running (pid 0000070d)
VBoxManage: error: Details: code VBOX_E_OBJECT_IN_USE (0x80bb000c), component DHCPServerWrap, interface IDHCPServer, callee nsISupports
VBoxManage: error: Context: "Start()" at line 482 of file VBoxManageNATNetwork.cpp
VBoxManage: error: Failed to start network
FATA[0013] VBoxManage error for command natnetwork start --netname natnet1 exit status 2 

```

Reply:
Thanks @dautovri for the feedback. Added explicit removal of the DHCP server and successfully verified that EVE restarts now works on both Linux and macOS.